Move users management to multi line

GretaD opened the pull request:
fix #17208 
The user management used to be like this:
 
![user_managn_old](https://user-images.githubusercontent.com/12728974/65420924-ad482c80-de02-11e9-8c11-6be76763c4a3.png)

Now it looks like this:

![user_mangn_new](https://user-images.githubusercontent.com/12728974/65420968-d1a40900-de02-11e9-9950-5d6be459687c.png)

This version is in read mode. The toggle to edit the display name will be added with another patch.
@nextcloud/designers what do you think? Any other suggestions related to design?

GretaD committed:
Move users management to multi line

Signed-off-by: Greta Doci <gretadoci@gmail.com>

ma12-co commented:
Looks good to me, I know there's a 'no uppercase' policy @jancborchardt :) but I think that it provides some nice hierarchy here. An alternative to that could be increasing the font size. 
The only thing I would do is vertically align those titles (the first is a bit off because of the sub heading). 
Also, you're pushing the compiled files, is that on purpose?

skjnldsv commented on settings/css/settings.scss:
```diff
@@ -1390,6 +1391,12 @@ doesnotexist:-o-prefocus, .strengthify-wrapper {
+			.subtitle,
+			.displayName
+			{
+				font-size: 10px;
```
Too small, we should aim for something a bit higher :)
Especially since we'll move to  15 px as default.

On #17208 screenshot, the size is the same, only the colour change ;)

GretaD replied:
![newnew](https://user-images.githubusercontent.com/12728974/65514205-27e57a80-dedd-11e9-8b99-e3f10238491e.png)

GretaD replied:
ah, i see, ok

jancborchardt commented:
> Looks good to me, I know there's a 'no uppercase' policy @jancborchardt :) but I think that it provides some nice hierarchy here. An alternative to that could be increasing the font size.

Hehe, so the point is that the focus should _not_ be on the table headings, but on the content. That is why they were lighter than the content text. The policy and the design does not come out of thin air. ;)

Uppercase text is quite bad for legibility as well as accessibility, and we shouldn’t start using it here on an otherwise unrelated change, especially not in table headings which are not a point of focus. :)

@GretaD could you change the headings to normal case, and a color like `color: var(--color-text-maxcontrast);`?

skjnldsv reviewed (changes requested):
Rebased and cleaned up :)

- [x] the menu on the end is not aligned anymore :)
   ![image](https://user-images.githubusercontent.com/14975046/69625908-d88c0980-1047-11ea-8d75-23c4d3108929.png)
- [x] Menu not vertically centered
   ![Capture d’écran_2019-11-26_12-26-39](https://user-images.githubusercontent.com/14975046/69626030-17ba5a80-1048-11ea-8381-231e6411a1a8.png)
- [ ] we should not show the password column if it's not in edit mode as nothing is displayed :)
   ![Capture d’écran_2019-11-26_12-27-38](https://user-images.githubusercontent.com/14975046/69626094-37518300-1048-11ea-9ccc-03f0b66208ef.png)
- [x] we can hide the userid when editing so only the displayname (editable) is shown in the middle
   ![Capture d’écran_2019-11-26_12-28-42](https://user-images.githubusercontent.com/14975046/69626154-59e39c00-1048-11ea-8522-1c86f17774d4.png)
- [x] Some slight mislignment of the multiselects
   ![Capture d’écran_2019-11-26_12-29-57](https://user-images.githubusercontent.com/14975046/69626227-813a6900-1048-11ea-8901-dd06cff6a969.png)
- [x] The quota bar doesn't have the tooltip with details like it had before (in readonly mode)

Otherwise great work!! It's looking very clean and much more understandable :)
:+1:

skjnldsv commented on apps/settings/src/components/UserList/UserRow.vue:
```diff
@@ -55,28 +75,31 @@
+							autocorrect="off"
+							autocapitalize="off"
+							spellcheck="false">
+						<input
```
- [ ] the inputs are not shown anymore :(

GretaD replied:
added back, can you please tell me if its ok now?

skjnldsv commented on apps/settings/src/components/UserList/UserRow.vue:
```diff
@@ -191,6 +214,11 @@
 		<div class="userActions">
 			<div v-if="OC.currentUser !== user.id && user.id !== 'admin' && !loading.all" class="toggleUserActions">
+				<Actions>
+					<ActionButton icon="icon-checkmark" @click="editing = false">
```
It just toggle the editing, it doesn't save", so the text should be something else :)
To save a property you still need to manually click the submit button next to the input you're changing (the multiselects automatically save on change)

GretaD replied:
I added "Done" is it better as an option?

skjnldsv commented on apps/settings/src/components/UserList/UserRow.vue:
```diff
@@ -516,10 +549,10 @@ export default {
-		 *
-		 * @param {string} gid Group id
-		 */
+		* Create a new group and add user to it
```
nope, you need the spaces here ;)
This is how comments are formatted :)

GretaD replied:
yep, fixed

skjnldsv commented on apps/settings/src/components/UserList/UserRowSimple.vue:
```diff
@@ -0,0 +1,164 @@
+	computed: {
+		userGroupsLabels() {
+			return this.userGroups
+				.map((groupObject) => groupObject.name)
```
nice! :+1:

GretaD commented:
@skjnldsv hey, we were discussing yesterday with Frank and Arthur about the "User Backend" column. Backend is very technical should we rename it something like Local maybe instead of Database? Or something else, less technical.

skjnldsv commented:
> Backend is very technical

I think it's hidden by default in the bottom left settings ;)

GretaD commented:
> > Backend is very technical
> 
> I think it's hidden by default in the bottom left settings ;)

i meant database not backend :) user backend is fine. the database we were discussing to call it smth like Local user or smth

skjnldsv commented:
Yes, it's the "user backend" (database, ldap...) It's hidden by default

jancborchardt commented:
> > > Backend is very technical
> > 
> > 
> > I think it's hidden by default in the bottom left settings ;)
> 
> i meant database not backend :) user backend is fine. the database we were discussing to call it smth like Local user or smth

To answer your question: Yes, changing the label from "Database" to "Local" seems good! :)

skjnldsv commented:
Ah, sorry I missed your point @GretaD :bowing_woman:

GretaD commented on lib/private/User/Database.php:
```diff
@@ -459,7 +459,7 @@ public function loginName2UserName($loginName) {
 	 */
 	public function getBackendName() {
-		return 'Database';
+		return 'Local';
```
im not sure where else should i change BackendName? Maybe also at UserManagment.php `if($user->getBackendClassName() === 'Database')` ?

skjnldsv replied:
I would keep this as is, but just change the displayed backend string. Here it will definitely blow up somewhere else.

skjnldsv replied:
cc @rullzer & @blizzz

rullzer replied:
Yep please keep it there....

skjnldsv commented:
Rebased to get latests master changes

skjnldsv commented:
Okay, fixed some stuff and cleaned up lots of errors.
Please have a look again @GretaD

skjnldsv commented:
/compile fixup /

karlitschek commented:
What is the state here? Would be great to have this in 18 :-)

karlitschek commented:
@skjnldsv @GretaD

skjnldsv commented:
> What is the state here? Would be great to have this in 18 :-)

hopefully we should be able to get this in yes ;)

GretaD commented:
> @skjnldsv @GretaD

yes, i will work on tests today and we can have a final review this week.

jancborchardt commented:
Some design review, if that is wanted ;)

![users](https://user-images.githubusercontent.com/925062/70540585-19cbff80-1b98-11ea-9a99-51cfdbce76b5.png)

- Nice on the username / display name column! :+1: 
- [ ] The "Password" column is shown but seems useless until you edit a user because it does not contain any info.
- [ ] Table heading only needs a border-bottom of 1px, the 2px is not necessary and we don’t do that elsewhere either.
- [ ] The "Group admin for" table heading wraps too much and looks broken. Should wrap to max 2 lines.
- [ ] Group memberships are not actually shown for me in the overview?
- [ ] Stuff is a bit squished, especially the controls on the right? Having the controls sticky on the right would be much better to have them easily reachable without horizontal scrolling (which is a huge pain)
- [ ] Same for the "Done" button, it could be stickied to the right. If you e.g. just want to edit a person’s email address, you might need to:
   1. Scroll to right
   2. Click edit button
   3. Do the edits
   4. Scroll to right again
   5. Click "Done" button
- [ ] There is some whitespace on the right that seems it should not be there?
    ![groups whitespace](https://user-images.githubusercontent.com/925062/70541152-16854380-1b99-11ea-9b9f-1a1572da10c9.png)

skjnldsv commented on apps/settings/src/components/UserList.vue:
```diff
@@ -455,18 +460,18 @@ export default {
-			 * In case the user directly loaded the user list within a group
-			 * the watch won't be triggered. We need to initialize it.
-			 */
+				 * In case the user directly loaded the user list within a group
```
Lots of issues with the comments tabs align here :)

GretaD replied:
my phpstarm is making my life miserable. I will fix it

skjnldsv replied:
^^

skjnldsv commented on tests/acceptance/features/bootstrap/UsersSettingsContext.php:
```diff
@@ -1,7 +1,7 @@
 
 /**
- * 
+ *
```
add yourself ;)

skjnldsv commented on tests/acceptance/features/bootstrap/UsersSettingsContext.php:
```diff
@@ -164,6 +164,14 @@ public static function editModeToggle($user) {
+	 */
+	public static function editModeOn($user) {
+		return Locator::forThe()->css(".row--editable")->
+		descendantOf(self::rowForUser($user))->
```
ah, it won't work as it looks for decendant of the row, but the class IS the row :thinking:

skjnldsv replied:
Let me see the options we have

skjnldsv replied:
Found, remove this line
```suggestion
```

skjnldsv commented on tests/acceptance/features/bootstrap/UsersSettingsContext.php:
```diff
@@ -155,6 +155,23 @@ public static function selectedSelectOption($cell, $user) {
+	 * @return Locator
+	 */
+	public static function editModeOn($user) {
+		return Locator::forThe()->css(".row--editable")->
```
```suggestion
		return Locator::forThe()->css("div.user-list-grid div.row.row--editable[data-id=$user]")->
```

skjnldsv replied:
And replace by this (we remove the check for decendant and match the row directly with the class :)

skjnldsv commented on tests/acceptance/features/users.feature:
```diff
@@ -1,80 +1,82 @@
-    And I open the User settings
-    And I see that the list of users contains the user user0
-    # disabled because we need the TAB patch: 
+	Scenario: assign user to a group
```
Please don't use the automated tabs option of phpstorm :see_no_evil:

GretaD replied:
:sob:

GretaD commented:
>     * [ ]  The "Password" column is shown but seems useless until you edit a user because it does not contain any info.

Removing it its not that user friendly since you have to wonder where to find the password to change it. We can add a placeholder or text to say smth like " click edit button to change password" or smth
>     * [ ]  Table heading only needs a border-bottom of 1px, the 2px is not necessary and we don’t do that elsewhere either.

Its `border-bottom-width: medium;` not border-bottom:2px. I changed it into `thin`.
> 
>     * [ ]  The "Group admin for" table heading wraps too much and looks broken. Should wrap to max 2 lines.

This is fixed, can you please check it out again, i cannot reproduce it anymore.
> 
>     * [ ]  Group memberships are not actually shown for me in the overview?

Yes, thats right, weird bug after the last changes pushed by @skjnldsv, will see whats wrong.
 
>     * [ ]  Same for the "Done" button, it could be stickied to the right. If you e.g. just want to edit a person’s email address, you might need to:
>       
>       1. Scroll to right
>       2. Click edit button
>       3. Do the edits
>       4. Scroll to right again
>       5. Click "Done" button

only for mobile, otherwise its fine, you dont need to scroll
>     * [ ]  There is some whitespace on the right that seems it should not be there?
>       ![groups whitespace](https://user-images.githubusercontent.com/925062/70541152-16854380-1b99-11ea-9b9f-1a1572da10c9.png)

this is also fixed, please recheck.

skjnldsv commented:
All rebased and ready to go.
Acceptance tests now passes locally, should be good!

Please review!

GretaD commented:
CI all greeeeennn yoohooo
